contracts/Core/BlockManager.sol

```diff
@@ -67,10 +68,10 @@ contract BlockManager is Initializable, ACL, BlockStorage, StateManager, IBlockM
         require(medians.length == assetManager.getNumActiveAssets(), "invalid block proposed");
 
         uint256 biggestInfluence = stakeManager.getInfluence(biggestInfluencerId);
-        uint8 numProposedBlocks = uint8(sortedProposedBlockIds[epoch].length);
+        if (sortedProposedBlockIds[epoch].length == 0) numProposedBlocks = 0;
```

Comment:
had to move from 
```
numProposedBlocks = uint8(sortedProposedBlockIds[epoch].length);
````
to having numProposedBlocks as state variable as 
if we replace the value,
length of sortedProposedBlockId don't change
so if new block comes in, it would have had same id 

for ex,
lets say we have 1,0
2 is proposed 
it replaces 1
so 2,1
now size is 2
so if we would have had length of array only as numProposedBlocks, new id would have been again 2

I think this problem existed earlier as well, with poping once we reach maxAltBlocks 
we have 
3,4,1,0,2

5 is proposed 

so 2 is popped
its now
3,4,5,1,0

now once new block is proposed here, with previous code,
its id would have been again 5, right ?